app/internal/app/dashboard_store.go

```diff
@@ -55,24 +80,79 @@ func (s *dashboardStore) GetAll(ctx context.Context) (donor, attorney, certifica
 		return nil, nil, nil, nil
 	}
 
-	var items []*page.Lpa
-	if err := s.dynamoClient.GetAllByKeys(ctx, searchKeys, &items); err != nil {
+	lpasOrProvidedDetails, err := s.dynamoClient.GetAllByKeys(ctx, searchKeys)
+	if err != nil {
 		return nil, nil, nil, err
 	}
 
-	for _, item := range items {
-		switch keyMap[item.ID] {
-		case actor.TypeDonor:
-			donor = append(donor, item)
-		case actor.TypeAttorney:
-			attorney = append(attorney, item)
-		case actor.TypeCertificateProvider:
-			certificateProvider = append(certificateProvider, item)
+	certificateProviderMap := make(map[string]page.LpaAndActorTasks)
+	attorneyMap := make(map[string]page.LpaAndActorTasks)
+
+	for _, item := range lpasOrProvidedDetails {
+		var ks keys
+		if err = attributevalue.UnmarshalMap(item, &ks); err != nil {
+			return nil, nil, nil, err
+		}
+
+		if ks.isLpa() {
+			lpa := &page.Lpa{}
+			if err := attributevalue.UnmarshalMap(item, lpa); err != nil {
+				return nil, nil, nil, err
+			}
+
+			switch keyMap[lpa.ID] {
+			case actor.TypeDonor:
+				donor = append(donor, page.LpaAndActorTasks{Lpa: lpa})
+			case actor.TypeAttorney:
+				attorneyMap[lpa.ID] = page.LpaAndActorTasks{Lpa: lpa}
+			case actor.TypeCertificateProvider:
+				certificateProviderMap[lpa.ID] = page.LpaAndActorTasks{Lpa: lpa}
+			}
+		}
+	}
+
+	for _, item := range lpasOrProvidedDetails {
```

Comment:
it would be good to reorder the results in the test to have one of the provided details come before the LPA, just because I think it looks very "obvious" that you can join these loops when in fact you should not

Reply:
Mmmm good point - I tried a few different ways to condense it down into one loop but it was getting gross looking and hard to follow. I'll add an additional test to spell this out.